tests/unit/test_sqs.py

```diff
@@ -46,3 +48,59 @@ def test_handle_string_max_receive_count_in_dead_letter_check():
     sqs_message = provider.SqsMessage(Message(), {})
     result = provider.SqsProvider()._dead_letter_check(queue, sqs_message, None)
     assert result is False
+
+
+def test_except_check_message_size():
+    message_body = "".join(("a" for _ in range(262145)))
+    result = False
+    try:
+        provider.check_message_size(message_body, MAXIMUM_MESSAGE_SIZE)
+    except provider.InvalidMessageContents:
+        result = True
+
+    assert result
+
+
+def test_noexc_check_message_size():
+    message_body = "a"
+    result = True
+    try:
+        provider.check_message_size(message_body, MAXIMUM_MESSAGE_SIZE)
+    except provider.InvalidMessageContents:
+        result = False
+
+    assert result
+
+
+def test_except_batch_message_size():
+    batch_message_size = 0
+    result = False
+    try:
+        for i in range(10):
+            batch_message_size += len((26215 * "b").encode("utf8"))
+            if batch_message_size > MAXIMUM_MESSAGE_SIZE:
+                error = (
+                    "Invalid batch message size found. Valid message size 262,144 bytes = 256KiB"
+                )
+                raise provider.InvalidMessageContents(error)
+    except provider.InvalidMessageContents:
+        result = True
+
+    assert result
```

Comment:
this seems like an odd/unnecessary test to me. what exactly are we testing from the provider?

Reply:
Yes. Not needed. It was just to isolate that section in the send_batch_message method under the failing case. I just wanted to make sure I had shown it was tested.

Reply:
an integration test will do!